app/api/chemotion/attachment_api.rb

```diff
@@ -443,6 +443,35 @@ def validate_uuid_format(uuid)
         end
       end
 
+      desc 'Regenerate edited spectra'
+      params do
+        requires :edited, type: Array[Integer]
+        optional :molfile, type: String
+      end
+      post 'regenerate_edited_spectrum' do
+        pm = to_rails_snake_case(params)
+        pm[:edited].each do |g_id|
+          att = Attachment.find(g_id)
+          next unless att
+          can_edit = writable?(att)
+          if can_edit
+            abs_path = att.abs_path
```

Comment:
NB: abs_path should not be used
it will have to be changed with the shrine upg
add a todo
```rb
# TODO: do not use abs_path
```